dockstore-webservice/src/test/java/io/dockstore/webservice/SimpleAuthenticatorTest.java

```diff
@@ -0,0 +1,76 @@
+package io.dockstore.webservice;
+
+import java.util.Optional;
+
+import com.google.api.services.oauth2.model.Userinfoplus;
+import io.dockstore.webservice.core.Token;
+import io.dockstore.webservice.core.User;
+import io.dockstore.webservice.jdbi.TokenDAO;
+import io.dockstore.webservice.jdbi.UserDAO;
+import io.dropwizard.auth.AuthenticationException;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import static org.junit.Assert.*;
```

Comment:
Configure IDE to avoid star imports

Reply:
1. Yes, I verified that a user is subsequently picked up by an explicit Google login. That is where I noticed the behavior with multiple logins that I started the email thread about.
2. Yes, the refresh token gets added when the user explicitly logs in afterward
3. The answer depends on what scopes the user granted for the token. FireCloud asks for email and profile scopes, so it works in that case. If for some reason we got a token from a whitelisted audience that didn't have those scopes, the call in `GoogleHelper.userinfoplusFromToken` would fail, which will lead to an unauthorized call.
4. That's odd, I have `Use single class import` selected in IDE, but I'm still getting the *. Will investigate.